rasa/core/utils.py

```diff
@@ -201,10 +201,19 @@ def _dump_yaml(obj: Dict, output: Union[Text, Path, StringIO]) -> None:
     yaml_writer.dump(obj, output)
 
 
-def dump_obj_as_yaml_to_file(filename: Union[Text, Path], obj: Dict) -> None:
-    """Writes data (python dict) to the filename in yaml repr."""
+def dump_obj_as_yaml_to_file(
+    filename: Union[Text, Path], obj: Dict, should_preserve_key_order: bool = False
```

Comment:
`obj` could also be of type list, str, float, bool (and probably more) . So maybe just use `Any`?